src/lib/utils/package-manifest.ts

```diff
@@ -36,6 +36,54 @@ export function loadPackageManifest(
     return packageJson as Record<string, unknown>;
 }
 
+export interface TypedocPackageManifestConfig {
+    entryPoint?: string | undefined;
+    displayName?: string | undefined;
+    readmeFile?: string | undefined;
+}
+
+/**
+ * Extracts typedoc specific config from a specified package manifest
+ */
+export function extractTypedocConfigFromPackageManifest(
+    logger: Logger,
+    packageJsonPath: string
+): TypedocPackageManifestConfig | undefined {
+    const packageJson = loadPackageManifest(logger, packageJsonPath);
+    if (!packageJson) {
+        return undefined;
+    }
+    if (
+        hasOwnProperty(packageJson, "typedoc") &&
+        typeof packageJson.typedoc == "object" &&
+        packageJson.typedoc
+    ) {
+        // TODO: we should consider another less manual approach in the future,
```

Comment:
There is `src/lib/utils/validation.ts`, this file was written before it existed and hasn't been updated to use it where it makes sense, see `src/test/utils/validation.ts` for usage examples.

Reply:
Great!! That's a really nice approach :). Updated!

Reply:
I think it makes sense to be strict today - there's no way to specify options here, so no plugins would expect to put stuff in this object.